thalassa/api.py

```diff
@@ -65,24 +65,102 @@ def get_elevation_dmap(trimesh: gv.TriMesh, show_grid: bool = False) -> hv.Overl
         overlay = tiles * elevation
     return overlay
 
-def get_timeseries(fid,source=None):
-    fdata=fid._dataset
-    sxi,syi=fdata['SCHISM_hgrid_node_x'].data,fdata['SCHISM_hgrid_node_y'].data
-    def get_timeseries_xy(x,y):
-        dist=abs(sxi+1j*syi-x-1j*y); nid=np.nonzero(dist==dist.min())[0][0]
-        mti=fdata['time'].data; myi=fdata['elev'].data[:,nid].copy()
-        ha=hv.Curve((mti,myi),'time','elevation')
-        return ha
-
-    #fixed station
-    xys=((3.0,37),(30,33)) #todo: read fixed stations from inputs
-    hs=hv.HoloMap({'P_{}'.format(i+1):get_timeseries_xy(*xy) for i,xy in enumerate(xys)})
-
-    #dynamic
-    hd=hv.DynamicMap(get_timeseries_xy,streams=[PointerXY(x=0,y=0,source=source)])
-    hd=hd.opts(color='k',line_width=2,line_dash='dotted')
-
-    #points
-    hp=hv.HoloMap({'P_{}'.format(i+1):hv.Points(xy).opts(color='r') for i,xy in enumerate(xys)})
-
-    return hd*hs.overlay(),hp.overlay()
+#----------------------------------------------------------------------------------------
+#time series
+#----------------------------------------------------------------------------------------
+#define a class to store data related to time series points
+class timeseries_data:
+      def __init__(self):
+         self.init=False
+      def clear(self):
+          self.init=False
+
+#function for extracting time series@(x,y) from data
+def extract_timeseries(x,y,sx,sy,data):
```

Comment:
In this function`data` is an `xr.Dataset` instance.  in the other functions `data` is a `timeseries_data` instance while the `xr.Dataset` instance is called `dataset`. I would suggest keeping the naming consistent.

Reply:
OK, I will change it later.